### PR TITLE
add error check of sensor_trigger_set()

### LINKED ISSUE #30189: Missing error check of function sensor_trigger_set()
Hi,
As described in the project API documentation, function sensor_trigger_set() is used to activate a sensor's trigger and set the trigger handler. It returns 0 if successful, negative errno code if failure. But as shown in the following codes, the return value is not error checked before it is assigned by another function.
https://github.com/zephyrproject-rtos/zephyr/blob/0a0404df29ed890cacadc94ecbd6e1e7f79c3d75/samples/sensor/sht3xd/src/main.c#L51-L61

But, the return values of the function from other call sites are all checked. See the following code,
https://github.com/zephyrproject-rtos/zephyr/blob/0a0404df29ed890cacadc94ecbd6e1e7f79c3d75/samples/sensor/sm351lt/src/main.c#L72-L76

### PR DESCRIPTION
**Pull request by Hxinrong**
### Contribution description
samples/sensor/sht3xd/src/main.c  ： add an error check of sensor_trigger_set()

### Signed-off
Signed-off-by: Xinrong Han hanxr19@mails.tsinghua.edu.cn

### Issue
Fixes #30189

### CODE REVIEW

File: samples/sensor/sht3xd/src/main.c [hunk 1]
```diff
@@ -50,6 +50,10 @@ void main(void)
 	}
 	if (rc == 0) {
 		rc = sensor_trigger_set(dev, &trig, trigger_handler);
+		if (rc != 0) {
+			printf("Failed to set trigger: %d\n", rc);
+			return;
+		}
 	}
```
Comment (avisconti) [inline samples/sensor/sht3xd/src/main.c:57]: Shouldn't the check be put here to catch also the errors of the previous API (sensor_attr_set called twice)?

**Review by avisconti — CHANGES_REQUESTED** [review]

File: samples/sensor/sht3xd/src/main.c (hunk 1 above)
Comment (pabigot) [reply samples/sensor/sht3xd/src/main.c:57]: Yes, the intent of the control flow as to check `rc == 0` after the chain of tests, and fail there, as is done in the similar chain within the while loop.

**Review by pabigot — CHANGES_REQUESTED** [review]
Please make this something like:

                if (rc != 0) {
                        printf("SHT3XD: failed threshold configuration: %d\n", rc);
                        return;
                }

and move it just before the `printf` showing the configured thresholds.

**Comment by avisconti** [comment]
> Please make this something like:
> 
> ```
>             if (rc != 0) {
>                     printf("SHT3XD: failed threshold configuration: %d\n", rc);
>                     break;
>             }
> ```
> 
> and move it just before the `printf` showing the configured threadsholds.

@pabigot
Seems reasonable

**Comment by pabigot** [comment]
Superseded by #31884

**Pull request closed without merging**